### PR TITLE
container: wait a bit before destroying non-stopped container

### PR DESCRIPTION
**Pull request by bergwolf**
Signal handling is async. If RemoveContainer is called right after
SignalProcess, it is possible that the container process is still
running. Let's wait a bit to see to give it some chance to exit
before calling destroy(). Otherwise destroy() might fail because
container process is still running.

cc @lifupan

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -188,11 +188,35 @@ func (c *container) deleteProcess(execID string) {
 	c.Unlock()
 }
 
+func (c *container) waitContainer() error {
+	for i := 0; i < 100; i++ {
+		if err := c.container.Signal(syscall.Signal(0), false); err != nil {
```
Comment (grahamwhaley) [inline agent.go]: Nice poll ;-)

File: agent.go [hunk 2]
```diff
@@ -188,11 +188,35 @@ func (c *container) deleteProcess(execID string) {
...
+	return fmt.Errorf("container is still running")
+}
+
 func (c *container) removeContainer() error {
+	s, err := c.container.Status()
+	if err != nil {
+		return err
+	}
+
+	// Wait a bit to see if container stops. This might happen if
```
Comment (devimc) [inline agent.go]: well, 10 seconds is not a bit

Comment (bergwolf) [reply agent.go]: Referencing [runc](https://github.com/opencontainers/runc/blob/master/delete.go#L18) :)

**Comment by bergwolf** [comment]
@sboeuf After discussing with @lifupan, I think a better approach is to make `WaitProcess` reentrant so that runtime can call it multiple times, like @laijs did in #131. I'll spin sometime to revive that one.

**Comment by sboeuf** [comment]
This PR can be closed as it's been superseded by another one.

**Pull request closed without merging**